modules/ROOT/pages/constraints/managing-constraints.adoc

```diff
@@ -0,0 +1,1598 @@
+----
+MATCH (book1:Book), (book2:Book)
+WHERE book1.title = book2.title AND NOT book1 = book2
+RETURN book1, book2
+----
+
+======
+
+
+.Create a relationship property existence constraint when conflicting relationships exist
+======
+
+
+.Create a constraint requiring all `WROTE` relationships to have a `language` property, when there already exists a relationship without a property named `language`
+[source, cypher, role=test-fail]
+----
+CREATE CONSTRAINT wrote_language FOR ()-[wrote:WROTE]-() REQUIRE wrote.language IS NOT NULL
+----
+
+In this case, the constraint cannot be created because it is in conflict with the existing graph.
+Remove or correct the offending relationships and then re-apply the constraint.
+
+.Error message
+[source, error]
+----
+Unable to create Constraint( type='RELATIONSHIP PROPERTY EXISTENCE', schema=()-[:WROTE {language}]-() ):
+Relationship(0) with type `WROTE` must have the property `language`. Note that only the first found violation is shown.
+----
+
+The constraint creation fails on the first offending relationship that is found.
+This does not guarantee that there are no other offending relationships in the graph.
+Therefore, all the data should be checked and cleaned up before re-attempting the constraint creation.
+
+
+.Find all offending relationships missing the property for the constraint above
+[source, cypher]
+----
+MATCH ()-[wrote:WROTE]-()
+WHERE wrote.language IS NULL
+RETURN wrote
+----
+
+======
+
+[[removing-an-existing-constrained-property-will-fail]]
+==== Removing existence constrained properties will fail
+
+.Remove a node property existence-constrained property
+======
+
+.Remove the `name` property from an existing `Author` node, given a property existence constraint on `:Author(name)`
+[source, cypher, role=test-fail]
+----
+MATCH (author:Author {name: 'Virginia Woolf'})
+REMOVE author.name
+----
+
+In this case, the property is not removed because it is in conflict with an existing constraint.
+
+.Error message
+[source, error]
+----
+Node(0) with label `Author` must have the property `name`
+----
+
+======
+
+
+.Remove a relationship property existence-constrained property
+======
+
+////
+[source, cypher, role=test-setup]
+----
+CREATE CONSTRAINT wrote_locations FOR ()-[wrote:WROTE]-() REQUIRE wrote.location IS NOT NULL
+----
+////
+
+.Remove the `location` property from an existing `WROTE` relationship, given a property existence constraint `:WROTE(location)`
+[source, cypher, role=test-fail]
+----
+MATCH (author:Author)-[wrote:WROTE]->(book:Book)
+REMOVE wrote.location
+----
+
+In this case, the property is not removed because it is in conflict with an existing constraint.
+
+.Error message
+[source, error]
+----
+Relationship(0) with type `WROTE` must have the property `location`
+----
+
+======
+
+[[create-key-and-uniqueness-constraint-on-same-schema-as-existing-index]]
+==== Creating key and uniqueness constraints on the same schema as an existing index will fail
```

Comment:
might be nice to mention this applies to all property uniqueness and key constraints